src/platform/plugins/shared/chart_expressions/expression_xy/public/components/tooltip/tooltip_actions.tsx

```diff
@@ -156,38 +209,81 @@ export const getTooltipActions = (
-                {
-                  table: xSeriesPoint.table,
-                  cells: [
-                    {
-                      row: xSeriesPoint.row,
-                      column: xSeriesPoint.column,
-                    },
-                  ],
-                },
-              ],
+            const { table } = xSeriesPoint;
+            const xColumn = getColumnByAccessor(xAccessor.toString(), table.columns);
+
+            // Get the field name and value for the Y axis
+            const selectedYValues = selectedValues.length ? selectedValues : [firstSeries];
+            const thresholdValues = selectedYValues.reduce((result, value) => {
+              const { yAccessor } = value.seriesIdentifier;
+              const yColumn = getColumnByAccessor(yAccessor.toString(), table.columns);
+              if (!yColumn || !yColumn.meta.sourceParams) return result;
+              const { sourceField } = yColumn.meta.sourceParams;
+              const yValue = value.value as number;
+              return {
+                ...result,
+                // If there is no sourceField, wrap the Y axis label in {curly braces} to let the user set the field name manually
+                [String(sourceField ?? `{${yColumn?.name ?? 'Y'}}`)]: yValue,
+              };
+            }, {});
+
+            // Get the time field name from the X axis for time vizzes, default to timestamp for non-time vizzes
+            const { sourceField: xSourceField } = xColumn?.meta?.sourceParams ?? {};
+            const timeField = isTimeViz && xSourceField ? String(xSourceField) : 'timestamp';
+
+            // If there are split accessors, get their values. For non-time vizzes, treat the X axis as a split accessor.
+            const splitValues: Record<
+              string,
+              Array<string | number | null | undefined>
+            > = isTimeViz || !hasXAxis
+              ? {}
+              : {
+                  // If there is no sourceField, wrap the X axis label in {curly braces} to let the user set the field name manually
+                  [String(xSourceField ?? `{${xColumn?.name ?? 'X'}}`)]: [xSeriesValue],
+                };
+            if (splitAccessors.size > 0) {
+              for (const [accessor, firstSplitValue] of splitAccessors) {
+                const splitColumn = table.columns.find((col) => col.id === accessor);
+                const { sourceField: splitSourceField } = splitColumn?.meta?.sourceParams ?? {};
+                if (!splitSourceField) continue;
+
+                const selectedSplitValues = selectedValues.length
+                  ? selectedValues.map((v) => v.seriesIdentifier.splitAccessors.get(accessor))
+                  : [firstSplitValue];
+                if (selectedSplitValues.length > 0)
+                  splitValues[String(splitSourceField)] = selectedSplitValues;
+              }
+            }
+
+            const query = table.meta?.type === 'es_ql' ? (table.meta.query as string) : null;
```

Comment:
I didn't do a code review but as I saw it: Let's use the constant here import { ESQL_TABLE_TYPE } from '@kbn/data-plugin/common';